File: src/python/finished/check_if_all_1s_are_at_least_length_k_places_away.py
from typing import List


# https://leetcode.com/problems/check-if-all-1s-are-at-least-length-k-places-away/


class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        last_seen = -k - 1
        for i, n in enumerate(nums):
            if n != 1:
                continue

            # n == 1
            if i - last_seen <= k:
                return False
            last_seen = i
        return True

File: src/rust/finished/check_if_all_1s_are_at_least_length_k_places_away.rs
impl Solution {
    pub fn k_length_apart(nums: Vec<i32>, k: i32) -> bool {
        let mut last_seen: i32 = -k - 1;
        for (i, &n) in nums.iter().enumerate() {
            if n != 1 {
                continue;
            }

            let i = i as i32;
            if i - last_seen <= k {
                return false;
            }

            last_seen = i;
        }

        true
    }
}
